subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskDefinitionIntegrationSpec.groovy

```diff
@@ -432,45 +303,26 @@ public class CustomPlugin implements Plugin<Project> {
         result.output.contains("got it")
     }
 
-    def "can construct a task with @Inject services and constructor args from a plugin"() {
+    @Requires([KOTLIN_SCRIPT])
+    def "can construct a task with @Inject services and constructor args via Project extension function"() {
         given:
-        file('buildSrc/src/main/java/CustomPlugin.java') << '''
-import org.gradle.api.*;
-import org.gradle.api.tasks.*;
-import org.gradle.workers.*;
-import javax.inject.Inject;
-
-public class CustomPlugin implements Plugin<Project> {
-    public static class CustomTask extends DefaultTask {
-        private final int number;
-        private final WorkerExecutor executor;
-
-        @Inject
-        public CustomTask(int number, WorkerExecutor executor) {
-            this.number = number;
-            this.executor = executor;
-        }
-
-        @TaskAction
-        void printIt() {
-            if (executor != null) {
-                System.out.println("got it " + number);
-            } else {
-                System.out.println(number + " NOT IT");
-            }
-        }
-    }
+        settingsFile << "rootProject.buildFileName = 'build.gradle.kts'"
+        file("build.gradle.kts") << """
+            import org.gradle.api.*
+            import org.gradle.api.tasks.*
+            import org.gradle.workers.WorkerExecutor
+            import javax.inject.Inject
 
-    @Override
-    public void apply(Project p) {
-        p.getTasks().create("myTask", CustomTask.class, 15);
-    }
-}
-'''
+            inline
+            fun <reified T : Task> Project.task(name: String, vararg arguments: Any?) =
```

Comment:
Why is this an extension function on `Project` and not on the `TaskContainer` (or whatever `Project.tasks` returns)?

Reply:
Hmm, I just looked at the kotlin DSL and the extension is really on `Project`...
I never saw/used that method, so I am a little bit confused now...

Reply:
resolved in slack conversation